src/main/java/org/opensearch/knn/indices/Model.java

```diff
@@ -148,25 +129,24 @@ public boolean equals(Object obj) {
         Model other = (Model) obj;
 
         EqualsBuilder equalsBuilder = new EqualsBuilder();
-        equalsBuilder.append(getModelMetadata(), other.getModelMetadata());
-        equalsBuilder.append(getModelBlob(), other.getModelBlob());
+        equalsBuilder.append(getModelID(), other.getModelID());
```

Comment:
We don't need EqualsBuilder.
```suggestion
                return other.getModelID().equals(this.getModelID());
```

Reply:
makes sense